resources/access.rb

```diff
@@ -80,29 +80,35 @@
 end
 
 action :create do
-  converge_if_changed do
-    config_resource_init
-    entry = config_resource.variables[:pg_hba].entry(new_resource.type, new_resource.database, new_resource.user, new_resource.address)
+  config_resource_init
+  entry = config_resource.variables[:pg_hba].entry(new_resource.type, new_resource.database, new_resource.user, new_resource.address)
 
-    if nil_or_empty?(entry)
+  if nil_or_empty?(entry)
+    converge_by("Add grant entry for #{new_resource.type}, #{new_resource.database}, #{new_resource.user}") do
       resource_properties = %i(type database user address auth_method auth_options comment).map { |p| [ p, new_resource.send(p) ] }.to_h.compact
       entry = PostgreSQL::Cookbook::AccessHelpers::PgHba::PgHbaFileEntry.create(**resource_properties)
       config_resource.variables[:pg_hba].add(entry)
-    else
+    end
+  elsif !entry.auth_method.eql?(new_resource.auth_method) || \
```

Comment:
You can pass properties to `converge_if_changed` to filter what triggers and update rather than the multiple `if`s. Combining that with `return` rather than `if/elsif` will be much cleaner.